packages/core/test/integration/query-interface/add-show-remove-index.test.ts

```diff
@@ -0,0 +1,304 @@
+'use strict';
+
+import { assert, expect } from 'chai';
+import { DataTypes, DatabaseError } from '@sequelize/core';
+import { getTestDialect, getTestDialectTeaser, sequelize } from '../support';
+
+const dialect = getTestDialect();
+const queryInterface = sequelize.queryInterface;
+
+describe(getTestDialectTeaser('QueryInterface#{add,show,removeIndex}'), () => {
+  describe('Standard Indexes', () => {
+    beforeEach(async () => {
+      await queryInterface.createTable('users', {
+        id: {
+          type: DataTypes.INTEGER,
+          primaryKey: true,
+          autoIncrement: true,
+        },
+        username: DataTypes.STRING,
+        first_name: DataTypes.STRING,
+        last_name: DataTypes.STRING,
+      });
+    });
+
+    it('should add, show and remove a non-unique index', async () => {
+      await queryInterface.addIndex('users', { name: 'unique_names', fields: ['username'], unique: false });
+
+      const indexes = await sequelize.queryInterface.showIndexes('users');
+      const indexCheck = indexes.find(index => index.name === 'unique_names');
+
+      expect(indexCheck).to.not.be.undefined;
+      assert(indexCheck);
+      expect(indexCheck.tableName).to.equal('users');
+      expect(indexCheck.name).to.equal('unique_names');
+      expect(indexCheck.unique).to.equal(false);
+      expect(indexCheck.fields).to.have.length(1);
+      expect(indexCheck.fields[0].name).to.equal('username');
+
+      await queryInterface.removeIndex('users', 'unique_names');
+      const indexesAfterRemoval = await sequelize.queryInterface.showIndexes('users');
+      const indexCheckAfterRemoval = indexesAfterRemoval.find(index => index.name === 'unique_names');
+
+      expect(indexCheckAfterRemoval).to.be.undefined;
+    });
+
+    it('should add, show and remove a unique index', async () => {
+      await queryInterface.addIndex('users', { name: 'unique_names', fields: ['first_name', 'last_name'], unique: true });
+
+      const indexes = await sequelize.queryInterface.showIndexes('users');
+      const indexCheck = indexes.find(index => index.name === 'unique_names');
+
+      expect(indexCheck).to.not.be.undefined;
+      assert(indexCheck);
+      expect(indexCheck.tableName).to.equal('users');
```

Comment:
prefer deep equal instead of checking individual properties?

That way we know if a dialect has extra properties and we can test that too

Same comment for the other tests

Reply:
Done.